client/tests/integration/unstable_network.rs

```diff
@@ -126,22 +114,25 @@ fn unstable_network(
             .expect("Failed to create asset.");
         account_has_quantity += quantity;
         thread::sleep(pipeline_time);
-    }
 
-    thread::sleep(pipeline_time);
+        //Then
+        iroha_client
+            .poll_request_with_period(
+                client::asset::by_account_id(account_id.clone()),
+                Configuration::pipeline_time(),
+                4,
+                |result| {
+                    result.iter().any(|asset| {
+                        asset.id().definition_id == asset_definition_id
+                            && *asset.value() == AssetValue::Quantity(account_has_quantity)
+                    })
+                },
+            )
+            .expect("Test case failure.");
 
-    //Then
-    iroha_client
-        .poll_request_with_period(
-            client::asset::by_account_id(account_id),
-            polling_period,
-            polling_max_attempts,
-            |result| {
-                result.iter().any(|asset| {
-                    asset.id().definition_id == asset_definition_id
-                        && *asset.value() == AssetValue::Quantity(account_has_quantity)
-                })
-            },
-        )
-        .expect("Test case failure.");
+        // Return all peers to normal function.
+        for f in &freezers {
+            f.store(0, Ordering::Relaxed);
+        }
+    }
```

Comment:
So the solution to test flakiness is that instead of having preset offline peers, we now choose them randomly for each transaction and reset at the end of each? Also just to be on the safe side, maybe it's better to: 
```suggestion
        // Return all peers to normal function.
        for f in &freezers {
            f.compare_exchange(1, 0, Ordering::Acquire, Ordering::Relaxed);
        }
    }
```

Reply:
The acquire would be meaningless because there is not any data or memory access to synchronize. This PR does nothing for any test flakiness. What it does is prepare the tests for us moving closer the whitepaper. It makes them more proper tests of sumeragi that the future changes won't be able to dodge.